modules/core/shared/src/main/scala/lucuma/core/model/SourceProfile.scala

```diff
@@ -267,4 +273,74 @@ object SourceProfile {
   /** @group Optics */
   val surfaceFluxDensityContinuum: Optional[SourceProfile, FluxDensityContinuumMeasure[Surface]] =
     surfaceSpectralDefinition.andThen(SpectralDefinition.fluxDensityContinuum[Surface])
-}
+
+  /**
+   * Returns the band and brightness value for the band closest to the given wavelength.
+   */
+  def extractBand[T](w: Wavelength, bMap: SortedMap[Band, BrightnessMeasure[T]]): Option[(Band, BrightnessValue, Units)] =
+    bMap.minByOption { case (b, _) =>
+      (w.toPicometers.value.value - b.center.toPicometers.value.value).abs
+    }.map(x => (x._1, x._2.value, x._2.units))
+
+  extension(sp: SourceProfile)
+    /**
+     * Returns the band and brightness of a source profile for the band closest to the given wavelength.
+     * It is only valid for source profiles with integrated or surface brightnesses.
+     * (Emission lines not supported)
+     */
+    def nearestBand(wavelength: Wavelength): Option[(Band, BrightnessValue, Units)] =
```

Comment:
Why an extension instead of a trait method?

Reply:
Could be in the trait, just wanted to keep the trait simple